app/avo/resources/user_resource.rb

```diff
@@ -36,7 +36,7 @@ class UserResource < Avo::BaseResource
   tabs style: :pills do
     tab "Auth" do
       field :encrypted_password, as: :password, visible: ->(_) { false }
-      field :mfa_seed, as: :text, visible: ->(_) { false }
+      field :totp_seed, as: :text, visible: ->(_) { false }
```

Comment:
Will this change break old audits that refer to user instances?

Reply:
Thanks for flagging this Samuel! Can you point me to read up on the "audits" you're speaking to here? I had understood Avo as being the admin panel for this application, so I thought this change would simply update the `mfa_seed` field in the admin panel to point to the `totp_seed `column in the db. 

It's unclear to me how this change affects a paper-trail/auditing system. Happy to investigate further if you can nudge me in the right direction. Thank you 🙏 !

Reply:
see the tests under `test/system/avo` -- basically, any actions undertaken in the admin dashboard have audits recorded that are a diff of all active record attributes, and then we use the avo resources to display the diff. My worry is that removing the `mfa_seed` field from the resource will mean that prior audits that touch that attribute will no longer have that change reflected in the audit's `show` view

Reply:
I recapped my [investigations](https://github.com/rubygems/rubygems.org/pull/3731#discussion_r1247363501) in @simi's PR since he's making a similar change.

Given this is the only blocker to shipping this PR, I'm going to propose taking the "safer" route first. Let's leave the deprecated/legacy field within `UserResource`. I'll add a comment noting it's legacy and whenever we align on approach on how to handle these legacy fields within the scope of the audit logs, we can revise this approach then.

If there aren't any objections, I'd like to ask @jenshenny to merge this in by end of week 🙏

Reply:
@segiddins I have checked how auditing is implmented and it seems to be too coupled to the codebase. We do initialize old/new object versions and use Avo resource to list the fields. That blocks various code changes from happening like class renaming (since auditable relation), column renaming/removing...

Instead we can just render the the changes list. That way it will not be related to current codebase.

WDYT?

Reply:
What do you mean, "just render the changes list"?

We intentionally re-use the avo resources so we properly display all the changed attributes, and don't show any attributes in the changes that we don't want exposed

Reply:
I mean there is GID and changes hash stored in DB.

```
irb(#<Avo::AuditedChangesRecordDiff::ShowComponent:0x00007fd5826b7448>):008:0> gid
=> "gid://gemcutter/User/5"
irb(#<Avo::AuditedChangesRecordDiff::ShowComponent:0x00007fd5826b7448>):009:0> changes
=> {"api_key"=>[nil, "6528fc7ae7c222275671e526f985a30f"], "updated_at"=>["2023-05-02T18:41:33.699Z", "2023-07-06T00:06:52.901Z"]}
```

Having this info loaded, it should be possible to render simple diff without initialising the User and assigning old/new attributes.

```
User 5
api_key changed from nil to "6528fc7ae7c222275671e526f985a30f" 
updated_at changed from "2023-05-02T18:41:33.699Z" to "2023-07-06T00:06:52.901Z"
```

This could be render even without `User` model being present or when `User` model has no `api_key` column since it is not coupled to the `User` model anymore.

Reply:
Thanks for raising this @simi. @jenshenny & I had briefly dug into alternative rendering options for the audits as well. Ultimately, decided iterating on how we render audits would be out of scope for this issue, so the "fix" in this PR was to ensure no regressions were introduced for rendering audits.

Admittedly, I don't have sufficient context or insight into the audit log's desired state. I don't have access to admin in prod either, so I'm not the target user for this part of the app. I'd defer to you all for what's important to show vs. hide.

To keep this PR moving forward, would y'all be comfortable with moving this discussion to a new issue and we can align on how best to handle rendering audits from there? I'll add a note that reminds us to revisit this current decision of continuing to define legacy fields. We can remove `mfa_seed` if needed once we align audit-rendering.

Let me know if that sounds like a fair plan 😊 !

Reply:
sounds good to me!